platform/representation/src/OneWayBinder.js

```diff
@@ -0,0 +1,111 @@
+         * @constructor
+         * @memberof platform/representation
+         * @param scope the Angular scope to watch
+         * @param attrs the relevant attributes, as passed into the `link`
+         *              function of the relevant directive
+         */
+        function OneWayBinder(scope, attrs) {
+            var self = this;
+
+            this.unwatches = [];
+            this.scope = scope;
+            this.parent = scope.$parent;
+            this.attrs = attrs;
+
+            // Detach any listeners from the parent
+            scope.$on('$destroy', function () {
+                self.unwatches.forEach(function (unwatch) {
+                    unwatch();
+                });
+            });
+        }
+
+        /**
+         * One-way bind an attribute. The value of the named attribute will
+         * be watched as an Angular expression in the parent scope; its
+         * value will be exposed in the child scope as a property of
+         * the same name.
+         * @param {string} attr the name of the attribute to watch
+         * @param {Function} [callback] a callback to invoke with new values
+         */
+        OneWayBinder.prototype.bind = function (attr, callback) {
+            this.alias(attr, attr, callback);
+        };
+
+        /**
+         * One-way bind an attribute. As `bind`, but allows the property
+         * name in the child scope used to expose these values to be
+         * specified as something different from the attribute name.
+         * @param {string} attr the name of the attribute to watch
+         * @param {string} property the name of the property to use in scope
+         * @param {Function} [callback] a callback to invoke with new values
+         */
+        OneWayBinder.prototype.alias = function (attr, property, callback) {
+            var scope = this.scope;
+
+            this.watch(attr, function expose(value) {
+                scope[property] = value;
+                if (callback) {
+                    callback(value);
+                }
+            });
+
+            // Expose in scope immediately, similar to scope: { attr: "=" }
+            // in a directive definition object.
+            scope[property] = this.parent.$eval(this.attrs[attr]);
+        };
+
+        /**
+         * Watch for changes in this attribute. The named attribute's value
+         * will be watched as an Angular expression in the parent scope,
+         * and the provided callback will be invoked with the value of that
+         * expression when changes occur.
+         * @param {string} attr the name of the attribute to watch
+         * @param {Function} callback the callback to invoke with new values
+         */
+        OneWayBinder.prototype.watch = function (attr, callback) {
+            var expr = this.attrs[attr];
+            this.unwatches.push(this.parent.$watch(
+                expr,
+                callback,
+                expr && expr[0] === '{'
```

Comment:
Perhaps a comment would be good here to explain the purpose of this check? It seems as though the result will be to do a shallow compare on object references and primitive types, and a deep compare on object literals (eg. when 'parameters' is specified). Is the value of an attribute on an object literal ever going to change though?